src/lsdb/loaders/hipscat/read_hipscat.py

```diff
@@ -28,6 +43,11 @@ def read_hipscat(
     }
     config = HipscatLoadingConfig(**config_args)
 
-    loader = HipscatCatalogLoader(path, config)
+    catalog_type_to_use = Catalog
```

Comment:
Why not just use a default arg value? or

```
if not catalog_type:
  catalog_type = Catalog
```

Reply:
It's to get the type checking to work. From what I can tell, I think because it's a TypeVar it doesn't like catalog_type being assigned to directly and gives errors, this was the only way I could get it to work. I did notice another Mypy error with this though so that's been updated now.